testsuite/MDAnalysisTests/coordinates/test_reader_api.py

```diff
@@ -111,50 +111,50 @@ def test_len(self):
         assert_equal(l, self.n_frames)
 
 
-class _Multi(object):
+class _Multi(_TestReader):
     n_frames = 10
     n_atoms = 10
     readerclass = AmazingMultiFrameReader
     reference = np.arange(10)
    
 
-class TestMultiFrameReader(_Multi, _TestReader):
-    def _check_slice(self, sl):
+class TestMultiFrameReader(_Multi):
+    def _check_slice(self, start, stop, step):
         """Compare the slice applied to trajectory, to slice of list"""
-        res = [ts.frame for ts in self.reader[sl]]
-        ref = self.reference[sl]
+        res = [ts.frame for ts in self.reader[start:stop:step]]
+        ref = self.reference[start:stop:step]
 
         assert_equal(res, ref)
     
-    def test_slice_1(self):
-        sl = slice(0, 10, 1)
-        self._check_slice(sl)
-
-    def test_slice_2(self):
-        sl = slice(0, 10, 2)
-        self._check_slice(sl)
-
-    def test_slice_3(self):
-        """Upper bound below traj length"""
-        sl = slice(0, 5, 2)
-        self._check_slice(sl)
-
-    def test_slice_4(self):
-        """Upper bound above traj length"""
-        sl = slice(0, 20, 2)
-        self._check_slice(sl)
-
-    def test_slice_5(self):
-        """Reverse order"""
-        sl = slice(0, 10, -1)
-        self._check_slice(sl)
-
-    def test_slice_IE_1(self):
+    def test_slices(self):
+        for start, stop, step in [
+                (None, None, None),
```

Comment:
I do not see the equivalent of `test_slice_4` with the upper bound above the traj length. Also, it would be nice to have a comment that tells what is being tested in each case (or at least the least trivial ones).